### PR TITLE
Dev auth

### PR DESCRIPTION
**Pull request by ultronjs**
-Feature Added in this PR
-Login
-Sign Up
-Homepage
-Firebase boilerplate code
Login Creds:-test@gmail.com password-123456
Will add deploy preview in next PR.Need to configure the firebase for Deploy preview url
Screenshot (377)
Screenshot (378)
Screenshot (381)

Screenshot (380)

### CODE REVIEW

File: src/pages/Dashboard.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  },[])
+
+  console.log(categories);
+  // async function handleLogout() {
+  //   setError("");
+
+  //   try {
+  //     await logout();
+  //     navigate("/login");
+  //   } catch {
```
Comment (ajit-sharmaGH) [inline src/pages/Dashboard.js:27]: You can remove if no need further comments

File: src/pages/Login.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    setLoading(false);
+  }
+
+  return (
+    <>
+      <NavBar />
+      <Container
+        className="d-flex align-items-center justify-content-center"
+        style={{ minHeight: "100vh", minWidth: "100vw" }}
```
Comment (ajit-sharmaGH) [inline src/pages/Login.js:36]: avoid using inline styling

**Review by ajit-sharmaGH — APPROVED** [review]
Proper names are used, 
like the useNavigate feature, 
You can merge it

File: src/pages/Signup.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    setLoading(false);
+  }
+
+  return (
+    <>
+      <NavBar />
+      <Container
+        className="d-flex align-items-center justify-content-center"
+        style={{ minHeight: "100vh", minWidth: "100vw" }}
```
Comment (Manju-sree) [inline src/pages/Signup.js:43]: try to avoid inline styling

**Review by Manju-sree — APPROVED** [review]
code looks clean,well structured and please avoid inline styling as much as possible in code. rest looks fine to me for merging this PR into dev.

File: src/pages/Signup.js (hunk 3 above)
Comment (ultronjs) [reply src/pages/Signup.js:43]: will resolve in upcoming pr

**Comment by ultronjs** [comment]
Merging after 2 Approvals

**Pull request merged**